core/chains/evm/logpoller/log_poller_test.go

```diff
@@ -0,0 +1,340 @@
+package logpoller
+
+import (
+	"context"
+	"database/sql"
+	"math/big"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind/backends"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/smartcontractkit/chainlink/core/chains/evm/client"
+	"github.com/smartcontractkit/chainlink/core/internal/gethwrappers/generated/log_emitter"
+	"github.com/smartcontractkit/chainlink/core/internal/testutils"
+	"github.com/smartcontractkit/chainlink/core/internal/testutils/pgtest"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/utils"
+)
+
+var (
+	EmitterABI, _ = abi.JSON(strings.NewReader(log_emitter.LogEmitterABI))
+)
+
+func GenLog(chainID *big.Int, logIndex int64, blockNum int64, blockHash string, topic1 []byte, address common.Address) Log {
+	return Log{
+		EvmChainId:  utils.NewBig(chainID),
+		LogIndex:    logIndex,
+		BlockHash:   common.HexToHash(blockHash),
+		BlockNumber: blockNum,
+		EventSig:    topic1,
+		Topics:      [][]byte{topic1},
+		Address:     address,
+		TxHash:      common.HexToHash("0x1234"),
+		Data:        []byte("hello"),
+	}
+}
+
+func assertDontHave(t *testing.T, start, end int, orm *ORM) {
+	for i := start; i < end; i++ {
+		_, err := orm.SelectBlockByNumber(int64(i))
+		assert.True(t, errors.Is(err, sql.ErrNoRows))
+	}
+}
+
+func assertHaveCanonical(t *testing.T, start, end int, ec *backends.SimulatedBackend, orm *ORM) {
+	for i := start; i < end; i++ {
+		blk, err := orm.SelectBlockByNumber(int64(i))
+		require.NoError(t, err)
+		chainBlk, err := ec.BlockByNumber(context.Background(), big.NewInt(int64(i)))
+		assert.Equal(t, chainBlk.Hash().Bytes(), blk.BlockHash.Bytes())
+	}
+}
+
+func TestUnit_LogPoller_PollAndSaveLogs(t *testing.T) {
```

Comment:
I find this naming convention very confusing. Why are we fighting against the go standards here? Go test names already have a very deliberate structure that is both semantic and functional. Inserting Unit_ and Integration_ prefixes everywhere undermines that.

Reply:
yeah I agree - @prashantkumar1982 should we re-evaluate? I get that it separates new and old tests, but doesn't seem worth the trouble

Reply:
Yes, the TestUnit_ was added only because we wanted to separate from old tests that weren't as per guidelines, with no plans of changing them. Especially for cases where we have a file with part old and part new tests. But ya, I guess practically we are not going to take a file with old tests, and add more tests to it in new format.  

However I did want to distinguish between unit-tests vs integration tests. It should be easy for someone going through a package code, to spot where the integration tests are, vs unit.

Reply:
> It should be easy for someone going through a package code, to spot where the integration tests are, vs unit.

Is this a realistic problem? Can you please point me to an example where this is not clear?

Typically Go package's have a test hierarchy structured like:
```go
TestFunc
TestFunc/case
TestType_method
TestType_method/case
TestType_Method/case/case
TestType_Integration
TestType_Integration/case
```

Can you help me understand specically how this improves clarity or utility? And why it is the _best_ way to do it? The loss of utility and resulting burden and drag on developers is significant:
```go
TestUnit_Func
TestUnit_Func/case
TestUnit_Type_method
TestUnit_Type_method/case
TestUnit_Type_Method/case/case
TestIntegeation_Type/case
```

Reply:
I think its clear just from the filename itself "integration_test" @prashantkumar1982? Also would apply more generally, anything that tests multiple components and thus is not named "fileundertest_test.go" is in some way an integration test.

Reply:
Adding qualifying words to disambiguate makes sense, but I think there can also be cases where a file/component _only_ has integration tests, and then it doesn't necesarily add value to include extra words. Also, we have to consider that there is not just one 'class' of integration tests, and we may want to reserve filename hints to signal build tags blocking external dependencies, like in the terra txm: https://github.com/smartcontractkit/chainlink/blob/develop/core/chains/terra/terratxm/txm_test.go#L1.